vault/plugin_catalog.go

```diff
@@ -303,9 +307,16 @@ func (c *PluginCatalog) isDatabasePlugin(ctx context.Context, pluginRunner *plug
 	// Attempt to run as database V5 or V6 plugin
 	v5Client, err := c.getPluginClient(ctx, nil, pluginRunner, config)
 	if err == nil {
-		multiplexingSupport := v5Client.MultiplexingSupport()
+		// At this point the pluginRunner does not know if multiplexing is
+		// supported or not. So we need to ask the plugin client itself.
+		multiplexingSupport, err := v5Client.MultiplexingSupport()
+		if err != nil {
+			return false, err
+		}
+
 		// Close the client and cleanup the plugin process
-		c.removeMultiplexedClient(pluginRunner.Name, v5Client.ID())
+		v5Client.Close()
```

Comment:
Probably want to log the error returned from this.

Reply:
resolved in: https://github.com/hashicorp/vault/pull/13896/commits/8adfd959af3486c89e46e5d58593608759ed11d1